### ISSUE TITLE
First argument to new HDWalletProvider() must be a mnemonic phrase, a single private key, or a list of private keys

### ISSUE THREAD

**Issue by ltfschoen** [PureStake/moonbeam-truffle-box#11]
Using the latest `"@truffle/hdwallet-provider": "^2.1.12",`, when I tried to run `truffle migrate --reset --compile-all --network moonbase` it gave me an error `First argument to new HDWalletProvider() must be a mnemonic phrase, a single private key, or a list of private keys.`, it now expects the private key is a Uint8Array with length 32 to be provided as an array. 
so provide a parameter `privateKeys` and type `string[]` as mentioned here https://github.com/trufflesuite/truffle/tree/develop/packages/hdwallet-provider

so i change it to 
```js
let args = {
   privateKeys: [privateKeyDev],
   providerOrUrl: 'http://localhost:9944/',
};
return new HDWalletProvider(args);

...

let args = {
   privateKeys: [privateKeyMoonbase],
   providerOrUrl: 'https://rpc.api.moonbase.moonbeam.network',
};
return new HDWalletProvider(args);
```

**Comment by eshaben** [comment]
hello! even with ` ^2.1.12` the private key as a string should work as it is a legacy parameter, but this raises a good point that we should update to the current accepted parameters (which also apply to the current version `^2.0.8`).

I'll submit a PR for it. thanks for bringing this up!
